### PR TITLE
test: Add Header permutations page

### PR DESCRIPTION
**Pull request by FlorianDr**
### Description

Add permutations page in preparation for refactoring and new functionality.

### Documentation changes

[*Do the changes include any API documentation changes?*]
- [ ] _Yes, this change contains documentation changes._
- [X] _No._

### Related Links

[*Attach any related links/pull request for this change*]

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- [ ] _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](CONTRIBUTING.md#public-apis)._
- [ ] _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](CONTRIBUTING.md#browsers-support)._
- [ ] _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- [ ] _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- [ ] _Changes are covered with new/existing unit tests?_
- [ ] _Changes are covered with new/existing integration tests?_
</details>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pages/header/permutations.page.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  {
+    variant: ['h1', 'h2', 'h3'],
+    info: [
+      <Link key="info" variant="info">
+        Info
+      </Link>,
+      null,
+    ],
+    description: ['description', null],
+    counter: ['counter', undefined],
```
Comment (timogasda) [inline pages/header/permutations.page.tsx:21]: minor: I think it'd be nicer to put more realistic values in here, like `10` or `3/10`

**Review by timogasda — APPROVED** [review]
One optional comment :)

File: pages/header/permutations.page.tsx (hunk 1 above)
Comment (FlorianDr) [reply pages/header/permutations.page.tsx:21]: If I get a fast approval, I might do it ;)

Comment (timogasda) [reply pages/header/permutations.page.tsx:21]: ⚡️

**Review by timogasda — APPROVED** [review]
LGTM!

**Pull request merged**